File: src/main/java/core/procedures/interop/BigIntegerType.java
package core.procedures.interop;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;
import core.utils.Utils;

import java.math.BigInteger;

public final class BigIntegerType extends AFn {

  public BigIntegerType() {
    super(new FnArgsBuilder().min(1).max(1).build());
  }

  @Override
  public boolean isPure() {
    return true;
  }

  @Override
  public String getName() {
    return "bigint";
  }

  @Override
  public BigInteger apply1(Object arg) {
    if (arg instanceof Number) {
      return Utils.toBigInteger((Number) arg);
    }
    return new BigInteger(arg.toString());
  }
}

File: src/main/java/core/procedures/interop/BigIntegerType.kt
package core.procedures.interop

import core.procedures.AFn
import core.procedures.FnArgsBuilder
import core.utils.Utils

import java.math.BigInteger

class BigIntegerType : AFn(FnArgsBuilder().min(1).max(1).build()) {

    override val isPure: Boolean
        get() = true

    override val name: String
        get() = "bigint"

    override fun apply1(arg: Any?): BigInteger {
        if (arg is Number) {
            return Utils.toBigInteger((arg as Number?)!!)
        }
        return BigInteger(arg!!.toString())
    }
}